File: app/src/main/java/com/asfoundation/wallet/ui/iab/AppcoinsRewardsBuyView.java
package com.asfoundation.wallet.ui.iab;

import androidx.annotation.Nullable;
import com.appcoins.wallet.bdsbilling.repository.entity.Purchase;
import io.reactivex.Observable;

interface AppcoinsRewardsBuyView {

  void finish(Purchase purchase);

  void showLoading();

  void hideLoading();

  void showNoNetworkError();

  Observable<Object> getOkErrorClick();

  Observable<Object> getSupportIconClick();

  Observable<Object> getSupportLogoClick();

  void close();

  void showGenericError();

  void finish(String uid);

  void errorClose();

  void finish(Purchase purchase, @Nullable String orderReference);

  void showTransactionCompleted();

  long getAnimationDuration();

  void lockRotation();
}

File: app/src/main/java/com/asfoundation/wallet/ui/iab/AppcoinsRewardsBuyView.kt
package com.asfoundation.wallet.ui.iab

import com.appcoins.wallet.bdsbilling.repository.entity.Purchase
import io.reactivex.Observable

interface AppcoinsRewardsBuyView {

  fun finish(purchase: Purchase)

  fun showLoading()

  fun hideLoading()

  fun showNoNetworkError()

  fun getOkErrorClick(): Observable<Any>

  fun getSupportIconClick(): Observable<Any>

  fun getSupportLogoClick(): Observable<Any>

  fun close()

  fun showError(message: Int?)

  fun finish(uid: String?)

  fun errorClose()

  fun showPaymentMethods()

  fun finish(purchase: Purchase, orderReference: String?)

  fun showTransactionCompleted()

  fun getAnimationDuration(): Long

  fun lockRotation()

  fun showVerification()
}